include/swift/Sema/ConstraintSystem.h

```diff
@@ -3249,6 +3249,12 @@ class ConstraintSystem {
   ConstraintLocator *
   getConstraintLocator(const ConstraintLocatorBuilder &builder);
 
+  /// Retrieves a locator that identifies where an overload ambiguity may be
+  /// present. In most cases this is a no-op, but may simplify locator e.g
+  /// coercions for ambiguities in the expression operand.
+  ConstraintLocator *
+  getConstraintLocatorForAmbiguity(ConstraintLocator *locator);
```

Comment:
Have you tried adjusting `getCalleeLocator` to detect presence of `CoercionOperand` and produce a version that is anchored on the operand?

Reply:
@xedin Most clients of `getCalleeLocator` don't want it to look through coercions, e.g for things like handling IUOs, default arguments, argument labels. We could add another `bool` parameter to handle that, but IMO it's clearer as a separate function.

Reply:
@hamishknight We added that element to indicate that the locator points to the coerced expression.

Reply:
@xedin Ah right I see, feels a little odd to call it a callee, but I guess that's fine to simplify things :)

Reply:
Just adjusted!
I was worried about this case: 
```
func fn63834_3() -> String {} // expected-note {{found candidate with type 'String'}}
func fn63834_3() -> Double {} // expected-note {{found candidate with type 'Double'}}

fn63834_3() as Int // expected-error {{no exact matches in call to global function 'fn63834_3'}}
```
I thought we had to consider adjusting the locator before `Solution::getCalleeLocator`. But we can just look recursively in that.